### PR TITLE
Selling process

### PR DESCRIPTION
**Pull request by EmirKapic**

### CODE REVIEW

File: frontend/src/pages/SellingProcess/ItemInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      >
+        <div className="flex flex-wrap gap-3">{images}</div>
+        <div className="absolute top-1/2 left-1/2 -translate-y-1/2 -translate-x-1/2 flex flex-col items-center gap-2">
+          <h3 className="text-purple font-bold text-2xl">
+            Click To Upload Photos
+          </h3>
+          <p className="text-lg">or just drag and drop</p>
+          <p className="text-lightgrey-200 text-sm">(Add at least 3 photos)</p>
+        </div>
+      </DragAndDrop>
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/ItemInfo.tsx:135]: Perhaps an `ImageUpload` component?

File: frontend/src/pages/SellingProcess/Prices.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            className="cursor-pointer flex flex-col"
+          >
+            Start date
+            <DatePicker
+              id={props.startDateId}
+              selected={props.startDate}
+              onChange={(date) => props.updateFields({ startDate: date })}
+              placeholderText="12/12/2024"
+              className="border border-silver py-2 px-5 cursor-pointer"
+              minDate={new Date()}
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/Prices.tsx:43]: What if I first select a start date, then select an end date and then change the start date to be after end date? We could add a `maxDate` for this case.

File: frontend/src/pages/SellingProcess/Prices.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          <label
+            htmlFor={props.endDateId}
+            className="cursor-pointer flex flex-col"
+          >
+            End date
+            <DatePicker
+              id={props.endDateId}
+              selected={props.endDate}
+              onChange={(date) => props.updateFields({ endDate: date })}
+              placeholderText="12/12/2024"
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/Prices.tsx:57]: Can we change this so it's easy to figure out whether it's DD/MM/YYYY or MM/DD/YYYY format

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 4]
```diff
@@ -0,0 +1,252 @@
+import Breadcrumb from "components/Common/Breadcrumb";
+import useMultistepForm from "hooks/useMultistepForm";
+import ItemInfo, { ImageFile } from "./ItemInfo";
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:3]: `ImageFile` is unused.

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  endDateId: "endDateId",
+};
+const shippingIds: ShippingInfoProps = {
+  addressId: "addressInput",
+  emailId: "Email",
+  cityId: "City",
+  zipId: "Zip Code",
+  countryId: "Country",
+  phoneId: "PhoneNumber",
+};
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:53]: Can we be consistent with the naming (case, spaces)

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+  }
+  const methods = useForm();
+
+  async function onFormSubmit(data: FieldValues): Promise<void> {
+    const uploadPromises = formState.imagePaths.map(
+      async (path) => await uploadFile(path),
+    );
+
+    const givenUrls = await Promise.all(uploadPromises);
+    console.log(givenUrls);
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:152]: 😮

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+      zipCode: data[shippingIds.zipId],
+      city: data[shippingIds.cityId],
+      country: data[shippingIds.countryId],
+      phoneNumber: data[shippingIds.phoneId],
+    };
+    const url = new UrlBuilder().products().url;
+    post<Product, NewProductRequest>(url, newProduct, {
+      headers: getAuthorizationHeaders(),
+    }).then(() => navigate("/"));
+    //Ovo promjeniti da bude neki interesantniji screen kao new auction sucessful
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:174]: Let `post` method add headers. We can just navigate to the product details page.

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+            onFormSubmit(data);
+          })}
+        >
+          {step}
+          <div className="flex justify-between">
+            <Button
+              type="secondary"
+              className={
+                btnClassName +
+                " hover:bg-lightgrey-200 hover:bg-opacity-40 duration-300"
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:197]: Don't we have a utility method for class name concatenation? Same below.

File: frontend/src/pages/SellingProcess/SellForm.tsx [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+                  type="primary"
+                  className={btnClassName}
+                  onClick={() => back()}
+                  formButtonType="button"
+                >
+                  back
+                </Button>
+              )}
+              {/* obviously, this should be done isLast ? onebtn : otherBtn but if i do it like that
+              due to some react shenanigans it submits form when it shouldnt*/}
```
Comment (farispoljcic-abh) [inline frontend/src/pages/SellingProcess/SellForm.tsx:215]: You could try adding them different keys, maybe that would help?

File: frontend/src/pages/SellingProcess/SellForm.tsx (hunk 7 above)
Comment (EmirKapic) [reply frontend/src/pages/SellingProcess/SellForm.tsx:174]: Fixed in a later branch.

File: backend/src/main/java/com/atlantbh/internship/AuctionApp/dtos/sell/NewProductRequest.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+import java.time.Instant;
+import java.util.List;
+
+public record NewProductRequest(@NonNull String title, long categoryId,
+                                long subcategoryId, @NonNull String description,
+                                @NonNull List<String> imageUrls, double startPrice,
+
+                                Instant startDate, Instant endDate,
+                                @NonNull String address, @NonNull String email,
```
Comment (farispoljcic-abh) [inline backend/src/main/java/com/atlantbh/internship/AuctionApp/dtos/sell/NewProductRequest.java:13]: Let's remove the email from the request like we agreed.

File: backend/src/main/java/com/atlantbh/internship/AuctionApp/utilities/ProductValidator.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
+package com.atlantbh.internship.AuctionApp.utilities;
+
+import com.atlantbh.internship.AuctionApp.dtos.sell.NewProductRequest;
+
+import java.time.Instant;
+
+//No tests for this one because all its doing is calling other validators, and those do have tests
```
Comment (farispoljcic-abh) [inline backend/src/main/java/com/atlantbh/internship/AuctionApp/utilities/ProductValidator.java:7]: Okay, just remove the comment.

File: frontend/src/services/fileManaging/FirebaseFileManager.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import app from "firebase/Firebase";
+import { getStorage, ref, uploadBytes, getDownloadURL } from "firebase/storage";
+import FileManager from "./FileManager";
+
+class FirebaseFileManager implements FileManager {
+  uploadFile(file: File): Promise<string> {
+    app;
```
Comment (farispoljcic-abh) [inline frontend/src/services/fileManaging/FirebaseFileManager.ts:7]: `app;` ?

**Review by farispoljcic-abh — APPROVED** [review]

**Pull request merged**